envi/archs/i386/emu.py

```diff
@@ -245,8 +245,13 @@ def executeOpcode(self, op):
         if meth is None:
             raise e_exc.UnsupportedInstruction(self, op)
 
+        # The behavior of the REP prefix is undefined when used with non-string instructions.
+        is_string_instruction = False
+        if op.mnem.startswith(("ins", "outs", "movs", "lods", "stos", "cmps", "scas")):
+            is_string_instruction = True
+
         rep_prefix = op.prefixes & PREFIX_REP_MASK
-        if rep_prefix and not self.getEmuOpt('i386:reponce'):
+        if rep_prefix and is_string_instruction and not self.getEmuOpt('i386:reponce'):
```

Comment:
hey @mr-tz 

thanks for your contribution!  this is conceptually a great thing and i want to push it forward.  i have a couple requests for changes tho.

this code is executed for every opcode that is emulated, not just the ones with the REP* prefixes, and string stuff is costly by nature.  can you do this:
* create a tuple of the INS_* constants (`INS_SCAS` for example) for easy integer comparisons (eg. `REP_OPCODES = (...)`)
* have the list created outside the function path (ie.  not as part of repeat execution, but only loaded at runtime) 
* remove the `is_string_instruction` stuff
* change the if to something like this:
```
if rep_prefix and op.opcode in REP_OPCODES and not self.getEmuOpt('i386:reponce'):
```

Reply:
thanks for the detailed feedback, see 9eb8415 for an update
hope that's what you've meant otherwise happy to modify this further

Reply:
+1